packages/app/contexts/TokenListContext.tsx

```diff
@@ -29,8 +37,15 @@ const TOKEN_LIST_BY_CHAIN_URL: { [chainId: number]: string } = {
   100: "https://tokens.honeyswap.org/",
 };
 
+const provider = new ethers.providers.JsonRpcProvider(
+  "https://rpc.gnosischain.com"
+);
```

Comment:
This is the thing I hate the most in the code I wrote, but I'm not 100% sure how to programmatically set this RPC provider here.

Tried using the [default provider](https://docs.ethers.org/v5/api/providers/#providers-getDefaultProvider) but it kept failing :c

Reply:
related to this, I this this code from @berteotti mint/gatekeeping logic PR can be used here as well

https://github.com/SwaprHQ/stackly-ui/pull/75/files#diff-b0ad06a11e66920646084475d9e320a6a38b6ec59ea4dcd4dd02f622c14e7ddaR32-R52